src/lib.rs

```diff
@@ -0,0 +1,198 @@
+//!
+//! - RPC APIs of a Substrate node: [`sc-rpc-api`]
+//! - CLI Options of a Substrate node: [`sc-cli`]
+//! - All of the consensus related crates provided by Substrate:
+//! 	- [`sc-consensus-aura`]
+//! 	- [`sc-consensus-babe`]
+//! 	- [`sc-consensus-grandpa`]
+//! 	- [`sc-consensus-beefy`]
+//! 	- [`sc-consensus-manual-seal`]
+//! 	- [`sc-consensus-pow`]
+//!
+//! Additional noteworthy external resources:
+//!
+//! - [Substrate Developer Hub](https://substrate.dev)
+//! - [Parity Tech's Documentation Hub](https://paritytech.github.io/)
+//! - [Frontier: Substrate's Ethereum Compatibility Library](https://paritytech.github.io/frontier/)
+//! - [Polkadot Wiki](https://wiki.polkadot.network/en/)
+//!
+//! Templates:
+//!
+//! - classic [`substrate-node-template`](https://github.com/substrate-developer-hub/substrate-node-template)
+//! - classic [cumulus-parachain-template](https://github.com/substrate-developer-hub/substrate-parachain-template)
+//! - [`extended-parachain-template`](https://github.com/paritytech/extended-parachain-template)
+//! - [`frontier-parachain-template`](https://github.com/paritytech/frontier-parachain-template)
+//!
+//! [polkadot]:
+//!     https://img.shields.io/badge/polkadot-E6007A?style=for-the-badge&logo=polkadot&logoColor=white
+//! [github]:
+//!     https://img.shields.io/badge/github-8da0cb?style=for-the-badge&labelColor=555555&logo=github
+//! [`sp-io`]: ../sp_io/index.html
+//! [`sp-api`]: ../sp_api/index.html
+//! [`sp-api`]: ../sp_api/index.html
+//! [`sc-client-db`]: ../sc_client_db/index.html
+//! [`sc-network`]: ../sc_network/index.html
+//! [`sc-rpc-api`]: ../sc_rpc_api/index.html
+//! [`sc-cli`]: ../sc_cli/index.html
+//! [`sc-consensus-aura`]: ../sc_consensus_aura/index.html
+//! [`sc-consensus-babe`]: ../sc_consensus_babe/index.html
+//! [`sc-consensus-grandpa`]: ../sc_consensus_grandpa/index.html
+//! [`sc-consensus-beefy`]: ../sc_consensus_beefy/index.html
+//! [`sc-consensus-manual-seal`]: ../sc_consensus_manual_seal/index.html
+//! [`sc-consensus-pow`]: ../sc_consensus_pow/index.html
+
+#![deny(rustdoc::broken_intra_doc_links)]
+#![deny(rustdoc::private_intra_doc_links)]
+
+#[cfg_attr(doc, aquamarine::aquamarine)]
+/// In this module, we explore substrate at a more depth. First, let's establish substrate being
+/// divided into a client and runtime.
+///
+/// ```mermaid
+/// graph TB
+/// subgraph Substrate
+/// 	direction LR
+/// 	subgraph Client
+/// 	end
+/// 	subgraph Runtime
+/// 	end
+/// end
+/// ```
+///
+/// The client and the runtime of course need to communicate. This is done through two concepts:
+///
+/// 1. Host functions: a way for the (Wasm) runtime to talk to the client. All host functions are
+///    defined in [`sp-io`]. For example, [`sp-io::storage`] are the set of host functions that
+///    allow the runtime to read and write data to the on-chain state.
+/// 2. Runtime APIs: a way for the client to talk to the Wasm runtime. Runtime APIs are defined
```

Comment:
A distinct difference between Runtime API and Host Function is that the Runtime API always creates a new WASM instance and does **not communicate** with an existing one.

Reply:
Seems more of an implementation detail rather than something worthy of being mentioned at this high level doc?